kafka-client/src/main/java/dev/responsive/kafka/internal/clients/ResponsiveKafkaClientSupplier.java

```diff
@@ -226,6 +227,20 @@ private String threadIdFromProducerConfig(final String clientId) {
     return match.group(1);
   }
 
+  /**
+   * @param clientId the restore consumer client id
+   * @return the extracted StreamThread id, of the form "StreamThread-n"
+   */
+  private String threadIdFromRestoreConsumerConfig(final String clientId) {
```

Comment:
why not just use the entire `clientId`?

Reply:
We need to be able to look up the store from all the different clients, so thread id is the finest granularity that's common to them all.